---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

## scisorseqr - a comparative analysis of alternative splicing patterns

```{r pressure, echo=FALSE, out.width = '25%', out.extra='style="float:left; padding:20px"'}
knitr::include_graphics("man/figures/scisorseqr.png")
```

scisorseqr is a linux based R-package for analyzing differential isoform expression in single cells. 
The methods are based on our preprint [Joglekar et al. (2020)]() and the [scISOrSeq workflow](https://www.nature.com/articles/nbt.4259?draft=marketing)

Any comparative studies of alternative splicing can be performed with scisorseqr. The
package includes functions for barcode deconvolution from fastqs,
integration with long read alignment tools, mapping and filtering of high confidence, 
full-length spliced reads, and some
handy tools to conduct differential expression analysis.

The tools are also applicable to long-read spatial transcriptomics, and can be used to resolve exon
expression at the [spatial](https://isoformatlas.com/) level

* * *

## Hardware / software requirements
The package has only been tested on a CentOS x86_64 machine. For alignment and mapping, we recommend

* [STARlong](https://github.com/alexdobin/STAR/) software installation for PacBio reads
* [Minimap2](https://github.com/lh3/minimap2) installation for Oxford Nanopore (or PacBio) reads
* samtools
* bedtools
* python version 3.7
 
## Installation

The easiest way to install scisorseqr is through [Github](https://github.com) with:

``` r
devtools::install_github('noush-joglekar/scisorseqr')
```

## Workflow

```{r flow, echo=FALSE, dev='pdf'}
# DiagrammeR::mermaid("graph TD;
#    A[Directory with fastq.gz files]-->B(GetBarcodes);
#    B-->E;
#    A-->C(STARalign);
#    C-->D(MapAndFilter);
#    D---E[InfoPerLongRead];
#    E-->F[IsoQuant];
#    E-->G[ExonQuant];
#    F-->H(DiffSplicingAnalysis);
#    G-->I(DiffSplicingAnalysis);
#    H-->J(Visualization);
#    I-->J((Visualization));
#    ")
```

``` {r prettyGraph, echo=FALSE, out.width = '60%'}
knitr::include_graphics("man/figures/README-flow-1.png")
```

These steps are available as functions in the package. For example, barcode deconvolution 
can be done using the following command

``` r 
library(scisorseqr)
GetBarcodes('FastqFiles/','userInput/BarcodeCluster_Assignments', concatenate=TRUE, 
  filterReads=FALSE, numProcesses=24)
```

## Support
We appreciate any and all inputs for improving scisorseqr. Feel free to send us an [email](mailto:anj2026@med.cornell.edu). 

